openff/bespokefit/schema/fitting.py

```diff
@@ -138,3 +153,18 @@ class BespokeOptimizationSchema(BaseOptimizationSchema):
         "By default bespoke torsion parameters (if requested) will be constructed for "
         "all non-terminal 'rotatable bonds'",
     )
+
+    expand_torsion_terms: bool = Field(
+        True,
+        description="If the optimization should first expand the number of k values "
+        "that should be fit for each torsion beyond what is in the initial force field.",
+    )
+    generate_bespoke_terms: bool = Field(
+        True,
+        description="If the optimized smirks should be bespoke to the target molecules.",
+    )
```

Comment:
I wonder if we should maybe wrap these in a new `BespokeSMIRKS` class (or something similar, I can't think of a good name right now!), and then move them directly into the optimization stage model so `parameters` becomes a union of `List[SMIRNOFFParameter]` and `BespokeSMIRKS`? That way we don't have an empty `parameters` field in addition to this other fields, and it also means each stage has more control over how the parameters are generated. E.g. you might want bespoke torsions but maybe the exact AM1BCCs.

Reply:
Great idea, I did also toy around with putting these settings in the hyperparameter objects but maybe that is too much control and we can just start with the optimization stage model. Maybe something like `SMIRKSettings` and I think I would prefer the settings to have their own field to keep things organised, validation would also be easier if the field is required rather than looping over the list to make sure it has been supplied.

Reply:
> That way we don't have an empty `parameters` field in


In most cases, the `parameters` field would be empty as we plan on generating these later in the workflow, but we could also allow users to pass their own list of parameters they wish to be optimised for example when they want to fit a subset of bonds and angles and not all of them which is the default behaviour. In this case, if the user passes the `SMIRKSettings.generate_bespoke_terms=False` and a list of parameters we would respect them and only fit those?